### ISSUE TITLE
Improve stubgenc type detection on PyBind11 properties

### ISSUE THREAD

**Issue by AntoinePrv** [python/mypy#8995]
Hello,

Thank you for the work done here on mypy. I don't know much about it yet but I believe the following would improve stub files generation for [Pybind11](https://pybind11.readthedocs.io/en/stable/) users.

## Current behavior
PyBind11 has the ability to bind C++ classes to Python objects.
While `stubgen` properly detects types for most functions/methods from the docstrings, types in properties (get, set) created from PyBind11, are reported as `Any`.

For instance, the following C++ code defines a class `Dog` and bind it to Python in a module named `python_example`.
```cpp
#include <pybind11/pybind11.h>

struct Dog {
  int age = 0;
  int n_friends = 0;
};

PYBIND11_MODULE(python_example, m) {
  pybind11::class_<Dog>(m, "Dog")
    .def(pybind11::init<>())  // Bind an __init__ method
    .def("get_age", [](Dog const& self){return self.age; })  // Bind a get_age method
    .def_readonly("age", &Dog::age)  // Bind a readonly property
    .def_readwrite("n_friends", &Dog::n_friends); // Bind a read and write property
}
```

Running `stubgen -m python_example` generates the following stub file
```python
from typing import Any

class Dog:
    def __init__(self) -> None: ...
    def get_age(self) -> int: ...
    @property
    def age(self) -> Any: ...
    @property
    def n_friends(self) -> Any: ...
    @n_friends.setter
    def n_friends(self, val: Any) -> None: ...
```
We can seed that while `stubgen` properly detect type for the `get_name` method, all properties are being created with `Any`.

## Expected behavior
It would be preferable if `stubgen` generated the following stub file
```python
from typing import Any

class Dog:
    def __init__(self) -> None: ...
    def get_age(self) -> int: ...
    @property
    def age(self) -> int: ...
    @property
    def n_friends(self) -> int: ...
    @n_friends.setter
    def n_friends(self, val: int) -> None: ...
```

## Docstring information
Pybind11 sets the following information in the docstrings
```python
In [1]: import python_example

In [2]: python_example.Dog.get_age.__doc__
Out[2]: 'get_age(self: python_example.Dog) -> int\n'

In [3]: python_example.Dog.age.__doc__
Out[3]: ''

In [4]: python_example.Dog.age.fget.__doc__
Out[4]: '(self: python_example.Dog) -> int\n'

In [5]: python_example.Dog.age.fset.__doc__

In [6]: python_example.Dog.n_friends.fget.__doc__
Out[6]: '(self: python_example.Dog) -> int\n'

In [7]: python_example.Dog.n_friends.fset.__doc__
Out[7]: '(self: python_example.Dog, arg0: int) -> None\n'

```

## Technical information
* mypy version: mypy 0.790+dev. 068594eb38fc550d834b7e61745177c423f2b224
* Python 3.8.2
* MacOs 10.15.5
* Compiler Apple clang version 11.0.3 (clang-1103.0.32.62) Target: x86_64-apple-darwin19.5.0

**Comment by ilevkivskyi** [comment]
I think we will accept a PR for this if someone volunteers.
